syntax = "proto3";
package pt.ulisboa.tecnico.tuplespaces.nameserver.contract;

message RegisterRequest {
    string service_name = 1;
    string qualifier = 2;
    string target = 3;
}

message RegisterResponse {
}

message LookupRequest {
    string service_name = 1;
}

message LookupResponse {
    repeated string targets = 1;   // target = ip:port
}

message RemoveRequest {
    string service_name = 1;
    string target = 2;             // target = ip:port
}

message RemoveResponse {
}

service NameServer {
    rpc register(RegisterRequest) returns (RegisterResponse);
    rpc lookup(LookupRequest) returns (LookupResponse);
    rpc remove(RemoveRequest) returns (RemoveResponse);
}

